.github/workflows/regression-report.yml

```diff
@@ -410,49 +403,27 @@ jobs:
     outputs:
       message: ${{ steps.out.outputs.message }}
     steps:
-      - name: Restore Master Checkstyle Checkout
-        uses: actions/cache@v4
-        with:
-          path: .ci-temp/checkstyle
-          key: "${{ env.CACHE_KEY_MASTER }}"
-
-      - name: Restore PR Checkout
-        uses: actions/cache@v4
+      - name: Restore cache
+        uses: actions/cache/restore@v4
         with:
           path: .
-          key: "${{ env.CACHE_KEY_PR }}"
+          key: ${{ env.regression-report-CACHE_KEY }}
+          fail-on-cache-miss: true
 
       - name: Restore config files from cache
-        uses: actions/cache@v4
+        uses: actions/cache/restore@v4
         with:
           path: |
             .ci-temp/*.xml
             .ci-temp/project.properties
-          key: ${{ env.CACHE_KEY_CONFIG_FILES }}
-
-      - name: Set upstream
-        run: |
-          bash
-          cd .ci-temp/checkstyle
-          MAIN_REPO_GIT_URL="https://github.com/checkstyle/checkstyle.git"
-          git remote add upstream "$MAIN_REPO_GIT_URL"
-          git fetch upstream
-          FORK_REPO_GIT_URL="https://github.com/$USER_LOGIN/checkstyle.git"
-          git remote add forked "$FORK_REPO_GIT_URL"
-          git fetch forked
+          key: ${{ env.regression-report-CACHE_KEY }}-config-files
 
       - name: Setup local maven cache
         uses: actions/cache@v4
         with:
           path: ~/.m2/repository
           key: checkstyle-maven-cache-${{ hashFiles('**/pom.xml') }}
```

Comment:
is it cache/restore ?

Reply:
this existed from earlier diff-report.yml. i just resued it.

Reply:
please update, right now  we are more knowledgeable on how to work with this action.
we should be explicit on store/restore

Reply:
moved to https://github.com/checkstyle/checkstyle/issues/15634